### ISSUE TITLE
Parsing simple Groovy file takes 10 s

### ISSUE THREAD

**Issue by robinst** [trishume/syntect#64]
The [UnmanagedNodeBackedModelMapTest.groovy file from the gradle sources](https://github.com/gradle/gradle/blob/aa8c200808824c61f2c46ed1eb948c759c9483c2/subprojects/model-core/src/test/groovy/org/gradle/model/UnmanagedNodeBackedModelMapTest.groovy) takes 10 seconds to parse, on current master (f03fa0f672ab0a25ec93cddba11a8a6b036605f9):

```
$ curl -LO "https://github.com/gradle/gradle/raw/aa8c200808824c61f2c46ed1eb948c759c9483c2/subprojects/model-core/src/test/groovy/org/gradle/model/UnmanagedNodeBackedModelMapTest.groovy"
$ cargo build --release --example syncat
$ time ./target/release/examples/syncat UnmanagedNodeBackedModelMapTest.groovy
...
    9.72s user 0.04s system 99% cpu 9.811 total
```

**Comment by trishume** [comment]
Darn, I bet this is more catastrophic backtracking.

**Comment by keith-hall** [comment]
This is interesting because there have basically been no changes to the Groovy syntax definition since the Sublime Text developer took it from TextMate, so I wonder about the performance in TextMate (and ST before the custom regex engine was implemented): https://github.com/sublimehq/Packages/commits/master/Groovy

**Comment by blach** [comment]
FWIW, I had the same problem in my own Sublime Text syntax parser I'm currently working on.
I could fix the problem by enabling `USE_COMBINATION_EXPLOSION_CHECK` in `regint.h` in Onigmo/Oniguruma.

**Comment by robinst** [comment]
Fixed in 2.0.1 by using a newer Onig which has the explosion check enabled by default, and handling it when we do matching.
